.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -3,14 +3,28 @@ Thank you for your pull request! Please review the requirements below
 and ensure your pull request has fulfilled all requirements outlined in the target package.
 -->
 
+## PR checklist
+
+For every PR, make sure that these statements are true:
+- [ ] Includes only changes relevant to the original ticket. Significant refactoring or formatting needs to be separated.
+- [ ] Doesn't contain unnecessary type casts and non-null asserts
+- [ ] Doesn't add `@ts-ignore`
+- [ ] Doesn't disable lints, particularly `no-any` lints
+- [ ] Doesn't use `useState` just for computation - use plain variables instead
+- [ ] Splits components into pure components that don't depend on external state or hooks.
+- [ ] Doesn't add components inside of components that depend on variables in the parent scope - separate them out
```

Comment:
Which point exactly? Here's a gpt4 explanation and rewrite: 
https://chat.openai.com/c/369ce4ad-c9ce-4ec2-9151-3ebd749c9a89

Reply:
I was referring to the last one, and I'm getting a "unable to load conversation" when I hit that URL, but anyway, I think my trouble comes from not understanding what's meant by "adding components inside of components". Do you mean this?

```tsx
const Foo: React.FC<{ ... }> = (outerProps) => {
  const Bar: React.FC<{ ... }> = (innerProps) => {
    // ...
  }

  // ...
}
```

Is that actually done anywhere in the code base?

Reply:
I see - that's exactly it. These cases are dangerous especially when the inner component closes over and therefore implicitly depends on some variables in the parent component scope. I eliminated some of those in the Checkout rewrite, but there are still some left in Manager, .e.g.: https://github.com/gitcoinco/grants-stack/blob/d9647b72b2a5f2dd22639f84863026565f33c22f/packages/round-manager/src/features/program/ViewProgramPage.tsx#226
You can probably see why this is a bad pattern - non-testable (or the tests for this small component end up in the bigger component which is also bad), implicit dependencies (instead of explicit via props or hooks), tight coupling etc.
The point of this check is to ensure that these are not introduced, and are properly separated out into their own component complete with isolated tests.

I'm more than open to a better wording for this, I know this isn't the best. gpt suggests:

_Avoid embedding components within other components if they depend on variables from the parent scope. Isolate such components for better modularity._

Reply:
I would simplify then. Since...

1. if an "inner component" doesn't depend on outer scope, then it doesn't need to be an inner component;
2. if an "inner component" does depend on outer scope, but we want that dependency removed, thereby making it an inner component that doesn't depend on outer scope, for which point 1) would apply...

Then how about just: _Avoid embedding components within other components_ ?

Reply:
That's great, thank you - I edited it to reflect this.